modules/core/src/v2/coins/cspr.ts

```diff
@@ -258,26 +265,64 @@ export class Cspr extends BaseCoin {
         throw new InvalidTransactionError('Error while trying to build transaction');
       }
       const id = Buffer.from(tx.casperTx.hash).toString('hex');
-      const transferId = accountLib.Cspr.Utils.getTransferId(tx.casperTx.session);
       const amount = accountLib.Cspr.Utils.getTransferAmount(tx.casperTx.session);
-      const toAddress = accountLib.Cspr.Utils.getTransferDestinationAddress(tx._deploy.session);
+      let transferId;
       const outputs: TransactionOutput[] = [];
-      // TODO(https://bitgoinc.atlassian.net/browse/STLX-677): define outputs for different types of tx
-      if (tx.type === accountLib.BaseCoin.TransactionType.Send) {
-        outputs.push({
-          address: toAddress,
-          amount,
-          coin: self.getChain(),
-        });
+      const operations: TransactionOperation[] = [];
+
+      switch (tx.type) {
+        case accountLib.BaseCoin.TransactionType.Send: {
+          transferId = accountLib.Cspr.Utils.getTransferId(tx.casperTx.session);
+          const toAddress = accountLib.Cspr.Utils.getTransferDestinationAddress(tx._deploy.session);
+          outputs.push({
+            address: toAddress,
+            amount,
+            coin: self.getChain(),
+          });
+          break;
+        }
+        case accountLib.BaseCoin.TransactionType.StakingLock: {
+          const validator = accountLib.Cspr.Utils.getValidatorAddress(tx._deploy.session);
+          operations.push({
+            type: accountLib.BaseCoin.TransactionType.StakingLock,
+            amount,
+            coin: self.getChain(),
+            validator: validator,
+          });
+          break;
+        }
+        case accountLib.BaseCoin.TransactionType.StakingUnlock: {
+          const validator = accountLib.Cspr.Utils.getValidatorAddress(tx._deploy.session);
+          operations.push({
+            type: accountLib.BaseCoin.TransactionType.StakingUnlock,
+            amount,
+            coin: self.getChain(),
+            validator: validator,
+          });
+          break;
+        }
+        default: {
+          throw new InvalidTransactionError('Error while trying to get transaction type');
```

Comment:
Here why are not covering wallet init, its ok?

Reply:
Yes, I spoke with Luis and he told me that it was not necessary.